sway/commands.c

```diff
@@ -278,10 +279,17 @@ list_t *execute_command(char *_exec, struct sway_seat *seat,
 			argv[i] = do_var_replacement(argv[i]);
 		}
 
-		if (!config->handler_context.using_criteria) {
+
+		if (!using_criteria) {
 			// The container or workspace which this command will run on.
-			struct sway_node *node = con ? &con->node :
-					seat_get_focus_inactive(seat, &root->node);
+			struct sway_node *node;
+			if(con) {
+				node = &con->node;
+				config->handler_context.node_overridden = true;
+			} else {
+				node = seat_get_focus_inactive(seat, &root->node);
+				config->handler_context.node_overridden = false;
```

Comment:
The internal state is only consistent so long as `node_overridden` is itself consistent with `handler_context.container`. For that to be the case, one has to remember to always set this flag appropriately after (or immediately before) a call to `set_config_node`.

Instead, I think `set_config_node` should take this bool as a parameter, to reduce chances for programmer error.

Reply:
Yes, that's probably a good idea.